test/relations.jl

```diff
@@ -287,152 +344,291 @@ end
+    test_partial_pressures(T, ρ, q_mixed, "mixed phase")
+
+    # Test 6: Partial pressure functions with thermodynamic state
+    T_test = FT(300)
+    ρ_test = FT(1.0)
+    q_test = PhasePartition(FT(0.01), FT(0.005), FT(0))
+    p_test = air_pressure(param_set, T_test, ρ_test, q_test)
+
+    # Create thermodynamic state and test consistency
+    e_int = internal_energy(param_set, T_test, q_test)
+    ts = PhaseNonEquil(param_set, e_int, ρ_test, q_test)
+
+    # Test partial pressures from thermodynamic state
+    p_vap_ts = partial_pressure_vapor(param_set, ts)
+    p_dry_ts = partial_pressure_dry(param_set, ts)
+
+    # Test consistency with direct computation
+    p_vap_direct = partial_pressure_vapor(param_set, p_test, q_test)
+    p_dry_direct = partial_pressure_dry(param_set, p_test, q_test)
+
+    @test p_vap_ts ≈ p_vap_direct
+    @test p_dry_ts ≈ p_dry_direct
+    @test p_vap_ts + p_dry_ts ≈ p_test  # Dalton's law
+
+    # Test 7: Partial pressure functions with AbstractPhaseDry (using PhasePartition approach)
+    T_dry = FT(300)
+    ρ_dry = FT(1.0)
+    q_dry = PhasePartition(FT(0))  # No water vapor
+    p_dry = air_pressure(param_set, T_dry, ρ_dry, q_dry)
+
+    # Test partial pressures for dry air using PhasePartition
+    p_vap_dry = partial_pressure_vapor(param_set, p_dry, q_dry)
+    p_dry_result = partial_pressure_dry(param_set, p_dry, q_dry)
+
+    # Validate dry air behavior: zero vapor pressure, total pressure equals dry air pressure
+    @test p_vap_dry === FT(0)  # No vapor pressure in dry air
+    @test p_dry_result ≈ p_dry  # Total pressure equals dry air pressure
+    @test p_vap_dry + p_dry_result ≈ p_dry  # Dalton's law for dry air
+
+    # Test 8: Partial pressure functions with some vapor present
+    T_moist = FT(300)
+    ρ_moist = FT(1.0)
+    q_moist = PhasePartition(FT(0.01))  # Some water vapor
+    p_moist = air_pressure(param_set, T_moist, ρ_moist, q_moist)
+
+    # Test partial pressures for moist air
+    p_vap_moist = partial_pressure_vapor(param_set, p_moist, q_moist)
+    p_dry_moist = partial_pressure_dry(param_set, p_moist, q_moist)
+
+    # Validate moist air behavior: positive vapor pressure, reduced dry air pressure
+    @test p_vap_moist > FT(0)  # Should have some vapor pressure
+    @test p_dry_moist < p_moist  # Dry air pressure should be less than total
+    @test p_vap_moist + p_dry_moist ≈ p_moist  # Dalton's law for moist air
+    @test p_dry_moist > FT(0)  # Dry air pressure should still be positive
+
+    # Test energy functions and their inverse temperature calculations
+    T = FT(300)
+    q_pt = PhasePartition(FT(0.02), FT(0.002), FT(0.002))
     e_kin = FT(11)
     e_pot = FT(13)
-    @test air_temperature(param_set, _cv_d * (T - _T_0) - _R_d * _T_0) === FT(T)
-    @test air_temperature(
-        param_set,
-        _cv_d * (T - _T_0) - _R_d * _T_0,
-        PhasePartition(FT(0)),
-    ) === FT(T)
+    e_int_dry = internal_energy_dry(param_set, T)
```

Comment:
Just a note. I think there was something useful in the previous test. The test would now fail to catch it if there is the same error in `internal_energy_dry` and `air_temperature`, for example, if they both use the wrong constant R_v * T_0 instead of R_d * T_0. I'm fine with this change though, as the better way to deal with this is to add a test for  the correctness of `internal_energy_dry` I think, which we don't have.

Reply:
Right, I just didn't like that we were manually recoding (i.e., copying and pasting) the definition from the source. I'd like the tests to be invariant under changes in the arbitrary constants, so what we should test is, e.g., consistency of energy and enthalpy. I'll add something.